k9mail/src/main/java/com/fsck/k9/activity/setup/outgoing/AccountSetupOutgoing.java

```diff
@@ -437,47 +375,35 @@ private void updateAuthPlainTextFromSecurityType(ConnectionSecurity securityType
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (resultCode == RESULT_OK) {
             if (Intent.ACTION_EDIT.equals(getIntent().getAction())) {
-                mAccount.save(Preferences.getPreferences(this));
+                presenter.updateAccount();
                 finish();
             } else {
-                AccountSetupOptions.actionOptions(this, mAccount, mMakeDefault);
+                AccountSetupOptions.actionOptions(this, presenter.getAccount(), mMakeDefault);
                 finish();
             }
         }
     }
 
     protected void onNext() {
         ConnectionSecurity securityType = getSelectedSecurity();
-        String uri;
-        String username = null;
-        String password = null;
-        String clientCertificateAlias = null;
-        AuthType authType = null;
-        if (mRequireLoginView.isChecked()) {
-            username = mUsernameView.getText().toString();
-
-            authType = getSelectedAuthType();
-            if (AuthType.EXTERNAL == authType) {
-                clientCertificateAlias = mClientCertificateSpinner.getAlias();
-            } else {
-                password = mPasswordView.getText().toString();
-            }
-        }
+        String username = mUsernameView.getText().toString();
+        String password = mPasswordView.getText().toString();
+        String clientCertificateAlias = mClientCertificateSpinner.getAlias();
+        AuthType authType = getSelectedAuthType();
 
         String newHost = mServerView.getText().toString();
         int newPort = Integer.parseInt(mPortView.getText().toString());
-        ServerSettings server = new ServerSettings(Type.SMTP, newHost, newPort, securityType, authType, username, password, clientCertificateAlias);
-        uri = Transport.createTransportUri(server);
-        mAccount.deleteCertificate(newHost, newPort, CheckDirection.OUTGOING);
-        mAccount.setTransportUri(uri);
-        AccountSetupCheckSettings.actionCheckSettings(this, mAccount, CheckDirection.OUTGOING);
+
+        boolean requireLogin = mRequireLoginView.isChecked();
+        presenter.next(username, password, clientCertificateAlias, newHost, newPort, securityType,
```

Comment:
Could make a model class for that, but I'm not sure that will help too much. I'd leave it as it is for now.